### PR TITLE
Resolve conflicts in otherlibs/dynlink/

### PR DESCRIPTION
**Pull request by mshinwell**
As subject.  Review commit by commit.

### CODE REVIEW

**Comment by mshinwell** [comment]
@xclerc maybe you could look at this, but let's discuss first.

File: ocaml/otherlibs/dynlink/byte/dynlink.ml [hunk 1]
```diff
@@ -196,31 +150,24 @@ module Bytecode = struct
...
         let events =
           if compunit.cu_debug = 0 then [| |]
           else begin
             seek_in ic compunit.cu_debug;
-            [| (Compression.input_value ic : Instruct.debug_event list) |]
+            [|
+              (* CR ocaml 5 compressed-marshal:
+              (Compression.input_value ic : Instruct.debug_event list)
+              *)
+              (Marshal.from_channel ic : Instruct.debug_event list)
```
Comment (xclerc) [inline ocaml/otherlibs/dynlink/byte/dynlink.ml:169]: How are we sure it is kept synchronized
with the code in `Emitcode`? Are the
tests enough?

File: ocaml/otherlibs/dynlink/byte/dynlink.ml [hunk 2]
```diff
@@ -196,31 +150,24 @@ module Bytecode = struct
           raise (DT.Error (Linking_error (file_name, new_error)))
         end;
         (* PR#5215: identify this code fragment by
-<<<<<<< HEAD
           digest of file contents + unit name.
           Unit name is needed for .cma files, which produce several code
           fragments. *)
+        (* XXX mshinwell: conflict resolution around here needs checking *)
```
Comment (xclerc) [inline ocaml/otherlibs/dynlink/byte/dynlink.ml:156]: (Now addressed?)

**Review by xclerc — APPROVED** [review]

File: ocaml/otherlibs/dynlink/dune [hunk 3]
```diff
@@ -36,10 +38,23 @@
...
+ (deps ../../parser.ml)
+ (action
+  (bash
+   "echo \"module MenhirLib = CamlinternalMenhirLib\" > parser.ml && cat ../../parser.ml >> parser.ml")))
+
+(rule
+ (targets camlinternalMenhirLib.ml)
+ (deps ../../parsing/camlinternalMenhirLib.ml)
+ (action
+  (copy %{deps} %{targets})))
```
[the remark this answers is not in the document]
Comment (mshinwell) [reply ocaml/otherlibs/dynlink/dune:52]: Yeah, this works so I'll leave it for now.

File: ocaml/otherlibs/dynlink/byte/dynlink.ml (hunk 1 above)
Comment (mshinwell) [reply ocaml/otherlibs/dynlink/byte/dynlink.ml:169]: Xavier and I discussed this, it seems ok.

**Pull request merged**